script/deploy_runner.sh

```diff
@@ -0,0 +1,34 @@
+#!/usr/bin/env bash
+
+# Copyright 2023 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+set -e
+
+rm -f github_runner.zip
+
+# Request a download URL for the artifact.
+echo "Requesting github runner charm download link..."
+DOWNLOAD_LOCATION=$(curl \
```

Comment:
I've learnt from Amanda about the -m option in curl, which is a timeout parameter for curl :D
>  -m, --max-time <fractional seconds>
              Maximum time in seconds that you allow each transfer to take.
              This is useful for preventing your batch jobs from hanging for
              hours due to slow networks or links going down.  Since 7.32.0,
              this option accepts decimal values, but the actual timeout will
              decrease in accuracy as the specified timeout increases in
              decimal precision.

Reply:
Great suggestion.
But this file will be removed, so this change is not needed.